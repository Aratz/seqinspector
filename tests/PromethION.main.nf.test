nextflow_pipeline {

    name "Test Workflow main.nf on PromethION data"
    script "../main.nf"
    tag "seqinspector"
    tag "PIPELINE"

    test("PromethION data test") {

        when {
            params {
                outdir = "$outputDir"
                input  = params.pipelines_testdata_base_path + "seqinspector/testdata/PromethION/samplesheet.csv"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("$outputDir/multiqc/groups/r10p41_e8p2_human_runs_jkw/multiqc_data/multiqc_citations.txt"),
                    path("$outputDir/multiqc/groups/r10p41_e8p2_human_runs_jkw/multiqc_data/multiqc_fastqc.txt"),
                    path("$outputDir/multiqc/groups/r10p41_e8p2_human_runs_jkw/multiqc_data/multiqc_general_stats.txt"),
                    path("$outputDir/multiqc/groups/r10p41_e8p2_human_runs_jkw/multiqc_data/multiqc_software_versions.txt"),

                    path("$outputDir/multiqc/multiqc_data/multiqc_citations.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_fastqc.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_software_versions.txt"),
                    ).match()
                },
            )
        }
    }
}
