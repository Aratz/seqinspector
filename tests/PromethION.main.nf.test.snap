{
    "PromethION data test": {
        "content": [
            "multiqc_citations.txt:md5,5f52d7a0141e4234c6069df9ef575c9a",
            "multiqc_fastqc.txt:md5,1a4b472e13cadc770832b0e20d1de7b0",
            "multiqc_general_stats.txt:md5,409cefc7f17f95d176ced6032bf8fb32"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-07T12:35:46.010641114"
    }
}